src/nodes/accessors/Normal.js

```diff
@@ -83,9 +83,13 @@ export const transformedNormalView = /*@__PURE__*/ ( Fn( ( builder ) => {
 
 	// Use getUV context to avoid side effects from nodes overwriting getUV in the context (e.g. EnvironmentNode)
 
-	return builder.context.setupNormal().context( { getUV: null } );
+	let node = builder.context.setupNormal().context( { getUV: null } );
 
-}, 'vec3' ).once() )().mul( faceDirection ).toVar( 'transformedNormalView' );
```

Comment:
@sunag I have moved the `mul( faceDirection )` statement into the `Fn` body so it's possible to evaluate the `flatShading` property. Does that have any side effects? What was the reason to add `mul( faceDirection )` after the `once()` call in the first place?

Reply:
It seems the E2E test of `webgpu_mrt` fails with this PR. The background of the normal pass looks different now.

Reply:
It seems that `Material` and `NodeMaterial` do not have the `.flatShading` property by default. I have changed the conditional logic to fix this.